deepinv/datasets/datagenerator.py

```diff
@@ -58,69 +95,168 @@ def __init__(
+                # Get the attribute name by stripping the split suffix
+                attr_name = member_name[: -len(split_suffix)]
+                # Register the ground truths
+                if attr_name == "x":
+                    x = member
+                # Register the measurements
+                elif attr_name == "y":
+                    if not marked_stacked:
+                        y = member
+                    else:
+                        warn(
+                            f"Dataset marked as stacked but found unstacked member {member_name}. There is probably an error with the dataset.",
+                            UserWarning,
+                            stacklevel=2,
+                        )
+                # Register parts of the stacked measurements and forward operator parameters
+                else:
+                    # Other members are understood as parameters except if they are stacked measurements
+                    registered_as_measurements = False
+
+                    # Register part of the stacked measurements
+                    # Example target values for member_name: "y0_train", "y1_train", ...
+                    if attr_name.startswith("y"):
+                        stacking_needle = attr_name[1:]
+                        # If needle is the canonical base 10 representation of a non-negative integer
+                        if (
+                            re.fullmatch(r"0|[1-9]\d*", stacking_needle, re.ASCII)
+                            is not None
+                        ):
+                            stacking_index = int(stacking_needle)
+                            # If the stacking index is in the target range
+                            if stacking_index in range(stacked):
+                                if marked_stacked:
+                                    registered_as_measurements = True
+                                    y[stacking_index] = member
+                                else:
+                                    warn(
+                                        f"Dataset not marked as stacked but found stacked member {member_name}. There is probably an error with the dataset.",
+                                        UserWarning,
+                                        stacklevel=2,
+                                    )
+                            else:
+                                warn(
+                                    f"Found stacked measurement member {member_name} with stacking index {stacking_index} outside of the expected range [0, {stacked}). There is probably an error with the dataset.",
+                                    UserWarning,
+                                    stacklevel=2,
+                                )
+
+                    # Register the forward operator parameters
+                    if params is not None and not registered_as_measurements:
+                        params[attr_name] = member
```

Comment:
also here we assume that all leftover members are params, right? I know this is the original implementation; it feels like we should have a warning when there are unused members but I guess it doesn't really work with this current implementation